daemon/models/cached_value.go

```diff
@@ -0,0 +1,63 @@
+package models
+
+import (
+	"sync"
+	"time"
+)
+
+// Thread safe data caching
+type CachedValue[T any] struct {
+	value       T
+	latestError error
+	cachedDate  time.Time
+	validity    time.Duration
+	updaterFn   func(self *CachedValue[T])
+	mu          *sync.Mutex
+}
+
+func NewCachedValue[T any](
+	value T,
+	err error,
```

Comment:
[small] maybe this could be named `latestError` also here? I think it would be easier to understand on the calling side